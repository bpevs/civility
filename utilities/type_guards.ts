import { EmptyArray, EmptyObj, Func, Nil, Obj } from '../types.ts'
import { parseRegExp } from './parse_regexp.ts'

export function isArray(item: unknown): item is unknown[] {
  return Array.isArray(item)
}

export function isBoolean(item: unknown): item is boolean {
  return typeof item === 'boolean'
}

export function isDate(item: unknown): item is Date {
  return item instanceof Date
}

export function isEmpty(item: unknown): item is EmptyArray | EmptyObj {
  if (isNil(item) || typeof item !== 'object') return false
  return Boolean(item && !Object.keys(item).length)
}

export function isError(item: unknown): item is Error {
  return item instanceof Error
}

export function isFunction(item: unknown): item is Func {
  return typeof item === 'function'
}

export function isNil(item: unknown): item is Nil {
  return item == null
}

export function isNull(item: unknown): item is null {
  return item === null
}

export function isNumber(item: unknown): item is number {
  return typeof item === 'number'
}

export function isObject(item: unknown): item is Obj<unknown> {
  if (isNull(item)) return false
  return isFunction(item) || typeof item === 'object'
}

export function isPromise(item: unknown): item is Promise<unknown> {
  return item instanceof Promise
}

export function isRegExp(item: unknown): item is RegExp {
  return item instanceof RegExp
}

export function isRegExpString(item: unknown): item is string {
  return isString(item) && Boolean(parseRegExp(item))
}

export function isSet(item: unknown): item is Set<unknown> {
  return item instanceof Set
}

export function isString(item: unknown): item is string {
  return typeof item === 'string'
}

export function isUndefined(item: unknown): item is undefined {
  return item === undefined
}
