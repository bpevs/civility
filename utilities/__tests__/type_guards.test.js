import { assertSnapshot } from '$std/testing/snapshot.ts'
import * as typeGuards from '../type_guards.ts'

const items = [
  's',
  0,
  1,
  '1',
  Infinity,
  null,
  undefined,
  [],
  [1],
  {},
  { 0: 1, 1: 2 },
  Number,
  new RegExp(/fsdf/),
  /fsdf/,
  1.5234,
  Math.pow(2, 53),
  new Error('hello'),
  NaN,
  Date.UTC(1, 1, 1),
  Promise.resolve(),
  new Set(),
  new Map(),
  '/myurlorsomething',
  '/myregex/aa',
  '/myregex/',
]

Object.keys(typeGuards).forEach((checkName) => {
  const check = typeGuards[checkName]
  try {
    Deno.test(check.name, async (t) => {
      await Promise.all(items.map((item) => {
        return assertSnapshot(t, `${check.name}(${item}): ${check(item)}`)
      }))
    })
  } catch (e) {
    console.error(e)
  }
})
