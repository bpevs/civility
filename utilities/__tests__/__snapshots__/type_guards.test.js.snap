export const snapshot = {};

snapshot[`isArray 1`] = `"isArray(s): false"`;

snapshot[`isArray 2`] = `"isArray(0): false"`;

snapshot[`isArray 3`] = `"isArray(1): false"`;

snapshot[`isArray 4`] = `"isArray(1): false"`;

snapshot[`isArray 5`] = `"isArray(Infinity): false"`;

snapshot[`isArray 6`] = `"isArray(null): false"`;

snapshot[`isArray 7`] = `"isArray(undefined): false"`;

snapshot[`isArray 8`] = `"isArray(): true"`;

snapshot[`isArray 9`] = `"isArray(1): true"`;

snapshot[`isArray 10`] = `"isArray([object Object]): false"`;

snapshot[`isArray 11`] = `"isArray([object Object]): false"`;

snapshot[`isArray 12`] = `"isArray(function Number() { [native code] }): false"`;

snapshot[`isArray 13`] = `"isArray(/fsdf/): false"`;

snapshot[`isArray 14`] = `"isArray(/fsdf/): false"`;

snapshot[`isArray 15`] = `"isArray(1.5234): false"`;

snapshot[`isArray 16`] = `"isArray(9007199254740992): false"`;

snapshot[`isArray 17`] = `"isArray(Error: hello): false"`;

snapshot[`isArray 18`] = `"isArray(NaN): false"`;

snapshot[`isArray 19`] = `"isArray(-2174774400000): false"`;

snapshot[`isArray 20`] = `"isArray([object Promise]): false"`;

snapshot[`isArray 21`] = `"isArray([object Set]): false"`;

snapshot[`isArray 22`] = `"isArray([object Map]): false"`;

snapshot[`isArray 23`] = `"isArray(/myurlorsomething): false"`;

snapshot[`isArray 24`] = `"isArray(/myregex/aa): false"`;

snapshot[`isArray 25`] = `"isArray(/myregex/): false"`;

snapshot[`isBoolean 1`] = `"isBoolean(s): false"`;

snapshot[`isBoolean 2`] = `"isBoolean(0): false"`;

snapshot[`isBoolean 3`] = `"isBoolean(1): false"`;

snapshot[`isBoolean 4`] = `"isBoolean(1): false"`;

snapshot[`isBoolean 5`] = `"isBoolean(Infinity): false"`;

snapshot[`isBoolean 6`] = `"isBoolean(null): false"`;

snapshot[`isBoolean 7`] = `"isBoolean(undefined): false"`;

snapshot[`isBoolean 8`] = `"isBoolean(): false"`;

snapshot[`isBoolean 9`] = `"isBoolean(1): false"`;

snapshot[`isBoolean 10`] = `"isBoolean([object Object]): false"`;

snapshot[`isBoolean 11`] = `"isBoolean([object Object]): false"`;

snapshot[`isBoolean 12`] = `"isBoolean(function Number() { [native code] }): false"`;

snapshot[`isBoolean 13`] = `"isBoolean(/fsdf/): false"`;

snapshot[`isBoolean 14`] = `"isBoolean(/fsdf/): false"`;

snapshot[`isBoolean 15`] = `"isBoolean(1.5234): false"`;

snapshot[`isBoolean 16`] = `"isBoolean(9007199254740992): false"`;

snapshot[`isBoolean 17`] = `"isBoolean(Error: hello): false"`;

snapshot[`isBoolean 18`] = `"isBoolean(NaN): false"`;

snapshot[`isBoolean 19`] = `"isBoolean(-2174774400000): false"`;

snapshot[`isBoolean 20`] = `"isBoolean([object Promise]): false"`;

snapshot[`isBoolean 21`] = `"isBoolean([object Set]): false"`;

snapshot[`isBoolean 22`] = `"isBoolean([object Map]): false"`;

snapshot[`isBoolean 23`] = `"isBoolean(/myurlorsomething): false"`;

snapshot[`isBoolean 24`] = `"isBoolean(/myregex/aa): false"`;

snapshot[`isBoolean 25`] = `"isBoolean(/myregex/): false"`;

snapshot[`isDate 1`] = `"isDate(s): false"`;

snapshot[`isDate 2`] = `"isDate(0): false"`;

snapshot[`isDate 3`] = `"isDate(1): false"`;

snapshot[`isDate 4`] = `"isDate(1): false"`;

snapshot[`isDate 5`] = `"isDate(Infinity): false"`;

snapshot[`isDate 6`] = `"isDate(null): false"`;

snapshot[`isDate 7`] = `"isDate(undefined): false"`;

snapshot[`isDate 8`] = `"isDate(): false"`;

snapshot[`isDate 9`] = `"isDate(1): false"`;

snapshot[`isDate 10`] = `"isDate([object Object]): false"`;

snapshot[`isDate 11`] = `"isDate([object Object]): false"`;

snapshot[`isDate 12`] = `"isDate(function Number() { [native code] }): false"`;

snapshot[`isDate 13`] = `"isDate(/fsdf/): false"`;

snapshot[`isDate 14`] = `"isDate(/fsdf/): false"`;

snapshot[`isDate 15`] = `"isDate(1.5234): false"`;

snapshot[`isDate 16`] = `"isDate(9007199254740992): false"`;

snapshot[`isDate 17`] = `"isDate(Error: hello): false"`;

snapshot[`isDate 18`] = `"isDate(NaN): false"`;

snapshot[`isDate 19`] = `"isDate(-2174774400000): false"`;

snapshot[`isDate 20`] = `"isDate([object Promise]): false"`;

snapshot[`isDate 21`] = `"isDate([object Set]): false"`;

snapshot[`isDate 22`] = `"isDate([object Map]): false"`;

snapshot[`isDate 23`] = `"isDate(/myurlorsomething): false"`;

snapshot[`isDate 24`] = `"isDate(/myregex/aa): false"`;

snapshot[`isDate 25`] = `"isDate(/myregex/): false"`;

snapshot[`isEmpty 1`] = `"isEmpty(s): false"`;

snapshot[`isEmpty 2`] = `"isEmpty(0): false"`;

snapshot[`isEmpty 3`] = `"isEmpty(1): false"`;

snapshot[`isEmpty 4`] = `"isEmpty(1): false"`;

snapshot[`isEmpty 5`] = `"isEmpty(Infinity): false"`;

snapshot[`isEmpty 6`] = `"isEmpty(null): false"`;

snapshot[`isEmpty 7`] = `"isEmpty(undefined): false"`;

snapshot[`isEmpty 8`] = `"isEmpty(): true"`;

snapshot[`isEmpty 9`] = `"isEmpty(1): false"`;

snapshot[`isEmpty 10`] = `"isEmpty([object Object]): true"`;

snapshot[`isEmpty 11`] = `"isEmpty([object Object]): false"`;

snapshot[`isEmpty 12`] = `"isEmpty(function Number() { [native code] }): false"`;

snapshot[`isEmpty 13`] = `"isEmpty(/fsdf/): true"`;

snapshot[`isEmpty 14`] = `"isEmpty(/fsdf/): true"`;

snapshot[`isEmpty 15`] = `"isEmpty(1.5234): false"`;

snapshot[`isEmpty 16`] = `"isEmpty(9007199254740992): false"`;

snapshot[`isEmpty 17`] = `"isEmpty(Error: hello): true"`;

snapshot[`isEmpty 18`] = `"isEmpty(NaN): false"`;

snapshot[`isEmpty 19`] = `"isEmpty(-2174774400000): false"`;

snapshot[`isEmpty 20`] = `"isEmpty([object Promise]): true"`;

snapshot[`isEmpty 21`] = `"isEmpty([object Set]): true"`;

snapshot[`isEmpty 22`] = `"isEmpty([object Map]): true"`;

snapshot[`isEmpty 23`] = `"isEmpty(/myurlorsomething): false"`;

snapshot[`isEmpty 24`] = `"isEmpty(/myregex/aa): false"`;

snapshot[`isEmpty 25`] = `"isEmpty(/myregex/): false"`;

snapshot[`isError 1`] = `"isError(s): false"`;

snapshot[`isError 2`] = `"isError(0): false"`;

snapshot[`isError 3`] = `"isError(1): false"`;

snapshot[`isError 4`] = `"isError(1): false"`;

snapshot[`isError 5`] = `"isError(Infinity): false"`;

snapshot[`isError 6`] = `"isError(null): false"`;

snapshot[`isError 7`] = `"isError(undefined): false"`;

snapshot[`isError 8`] = `"isError(): false"`;

snapshot[`isError 9`] = `"isError(1): false"`;

snapshot[`isError 10`] = `"isError([object Object]): false"`;

snapshot[`isError 11`] = `"isError([object Object]): false"`;

snapshot[`isError 12`] = `"isError(function Number() { [native code] }): false"`;

snapshot[`isError 13`] = `"isError(/fsdf/): false"`;

snapshot[`isError 14`] = `"isError(/fsdf/): false"`;

snapshot[`isError 15`] = `"isError(1.5234): false"`;

snapshot[`isError 16`] = `"isError(9007199254740992): false"`;

snapshot[`isError 17`] = `"isError(Error: hello): true"`;

snapshot[`isError 18`] = `"isError(NaN): false"`;

snapshot[`isError 19`] = `"isError(-2174774400000): false"`;

snapshot[`isError 20`] = `"isError([object Promise]): false"`;

snapshot[`isError 21`] = `"isError([object Set]): false"`;

snapshot[`isError 22`] = `"isError([object Map]): false"`;

snapshot[`isError 23`] = `"isError(/myurlorsomething): false"`;

snapshot[`isError 24`] = `"isError(/myregex/aa): false"`;

snapshot[`isError 25`] = `"isError(/myregex/): false"`;

snapshot[`isFunction 1`] = `"isFunction(s): false"`;

snapshot[`isFunction 2`] = `"isFunction(0): false"`;

snapshot[`isFunction 3`] = `"isFunction(1): false"`;

snapshot[`isFunction 4`] = `"isFunction(1): false"`;

snapshot[`isFunction 5`] = `"isFunction(Infinity): false"`;

snapshot[`isFunction 6`] = `"isFunction(null): false"`;

snapshot[`isFunction 7`] = `"isFunction(undefined): false"`;

snapshot[`isFunction 8`] = `"isFunction(): false"`;

snapshot[`isFunction 9`] = `"isFunction(1): false"`;

snapshot[`isFunction 10`] = `"isFunction([object Object]): false"`;

snapshot[`isFunction 11`] = `"isFunction([object Object]): false"`;

snapshot[`isFunction 12`] = `"isFunction(function Number() { [native code] }): true"`;

snapshot[`isFunction 13`] = `"isFunction(/fsdf/): false"`;

snapshot[`isFunction 14`] = `"isFunction(/fsdf/): false"`;

snapshot[`isFunction 15`] = `"isFunction(1.5234): false"`;

snapshot[`isFunction 16`] = `"isFunction(9007199254740992): false"`;

snapshot[`isFunction 17`] = `"isFunction(Error: hello): false"`;

snapshot[`isFunction 18`] = `"isFunction(NaN): false"`;

snapshot[`isFunction 19`] = `"isFunction(-2174774400000): false"`;

snapshot[`isFunction 20`] = `"isFunction([object Promise]): false"`;

snapshot[`isFunction 21`] = `"isFunction([object Set]): false"`;

snapshot[`isFunction 22`] = `"isFunction([object Map]): false"`;

snapshot[`isFunction 23`] = `"isFunction(/myurlorsomething): false"`;

snapshot[`isFunction 24`] = `"isFunction(/myregex/aa): false"`;

snapshot[`isFunction 25`] = `"isFunction(/myregex/): false"`;

snapshot[`isNil 1`] = `"isNil(s): false"`;

snapshot[`isNil 2`] = `"isNil(0): false"`;

snapshot[`isNil 3`] = `"isNil(1): false"`;

snapshot[`isNil 4`] = `"isNil(1): false"`;

snapshot[`isNil 5`] = `"isNil(Infinity): false"`;

snapshot[`isNil 6`] = `"isNil(null): true"`;

snapshot[`isNil 7`] = `"isNil(undefined): true"`;

snapshot[`isNil 8`] = `"isNil(): false"`;

snapshot[`isNil 9`] = `"isNil(1): false"`;

snapshot[`isNil 10`] = `"isNil([object Object]): false"`;

snapshot[`isNil 11`] = `"isNil([object Object]): false"`;

snapshot[`isNil 12`] = `"isNil(function Number() { [native code] }): false"`;

snapshot[`isNil 13`] = `"isNil(/fsdf/): false"`;

snapshot[`isNil 14`] = `"isNil(/fsdf/): false"`;

snapshot[`isNil 15`] = `"isNil(1.5234): false"`;

snapshot[`isNil 16`] = `"isNil(9007199254740992): false"`;

snapshot[`isNil 17`] = `"isNil(Error: hello): false"`;

snapshot[`isNil 18`] = `"isNil(NaN): false"`;

snapshot[`isNil 19`] = `"isNil(-2174774400000): false"`;

snapshot[`isNil 20`] = `"isNil([object Promise]): false"`;

snapshot[`isNil 21`] = `"isNil([object Set]): false"`;

snapshot[`isNil 22`] = `"isNil([object Map]): false"`;

snapshot[`isNil 23`] = `"isNil(/myurlorsomething): false"`;

snapshot[`isNil 24`] = `"isNil(/myregex/aa): false"`;

snapshot[`isNil 25`] = `"isNil(/myregex/): false"`;

snapshot[`isNull 1`] = `"isNull(s): false"`;

snapshot[`isNull 2`] = `"isNull(0): false"`;

snapshot[`isNull 3`] = `"isNull(1): false"`;

snapshot[`isNull 4`] = `"isNull(1): false"`;

snapshot[`isNull 5`] = `"isNull(Infinity): false"`;

snapshot[`isNull 6`] = `"isNull(null): true"`;

snapshot[`isNull 7`] = `"isNull(undefined): false"`;

snapshot[`isNull 8`] = `"isNull(): false"`;

snapshot[`isNull 9`] = `"isNull(1): false"`;

snapshot[`isNull 10`] = `"isNull([object Object]): false"`;

snapshot[`isNull 11`] = `"isNull([object Object]): false"`;

snapshot[`isNull 12`] = `"isNull(function Number() { [native code] }): false"`;

snapshot[`isNull 13`] = `"isNull(/fsdf/): false"`;

snapshot[`isNull 14`] = `"isNull(/fsdf/): false"`;

snapshot[`isNull 15`] = `"isNull(1.5234): false"`;

snapshot[`isNull 16`] = `"isNull(9007199254740992): false"`;

snapshot[`isNull 17`] = `"isNull(Error: hello): false"`;

snapshot[`isNull 18`] = `"isNull(NaN): false"`;

snapshot[`isNull 19`] = `"isNull(-2174774400000): false"`;

snapshot[`isNull 20`] = `"isNull([object Promise]): false"`;

snapshot[`isNull 21`] = `"isNull([object Set]): false"`;

snapshot[`isNull 22`] = `"isNull([object Map]): false"`;

snapshot[`isNull 23`] = `"isNull(/myurlorsomething): false"`;

snapshot[`isNull 24`] = `"isNull(/myregex/aa): false"`;

snapshot[`isNull 25`] = `"isNull(/myregex/): false"`;

snapshot[`isNumber 1`] = `"isNumber(s): false"`;

snapshot[`isNumber 2`] = `"isNumber(0): true"`;

snapshot[`isNumber 3`] = `"isNumber(1): true"`;

snapshot[`isNumber 4`] = `"isNumber(1): false"`;

snapshot[`isNumber 5`] = `"isNumber(Infinity): true"`;

snapshot[`isNumber 6`] = `"isNumber(null): false"`;

snapshot[`isNumber 7`] = `"isNumber(undefined): false"`;

snapshot[`isNumber 8`] = `"isNumber(): false"`;

snapshot[`isNumber 9`] = `"isNumber(1): false"`;

snapshot[`isNumber 10`] = `"isNumber([object Object]): false"`;

snapshot[`isNumber 11`] = `"isNumber([object Object]): false"`;

snapshot[`isNumber 12`] = `"isNumber(function Number() { [native code] }): false"`;

snapshot[`isNumber 13`] = `"isNumber(/fsdf/): false"`;

snapshot[`isNumber 14`] = `"isNumber(/fsdf/): false"`;

snapshot[`isNumber 15`] = `"isNumber(1.5234): true"`;

snapshot[`isNumber 16`] = `"isNumber(9007199254740992): true"`;

snapshot[`isNumber 17`] = `"isNumber(Error: hello): false"`;

snapshot[`isNumber 18`] = `"isNumber(NaN): true"`;

snapshot[`isNumber 19`] = `"isNumber(-2174774400000): true"`;

snapshot[`isNumber 20`] = `"isNumber([object Promise]): false"`;

snapshot[`isNumber 21`] = `"isNumber([object Set]): false"`;

snapshot[`isNumber 22`] = `"isNumber([object Map]): false"`;

snapshot[`isNumber 23`] = `"isNumber(/myurlorsomething): false"`;

snapshot[`isNumber 24`] = `"isNumber(/myregex/aa): false"`;

snapshot[`isNumber 25`] = `"isNumber(/myregex/): false"`;

snapshot[`isObject 1`] = `"isObject(s): false"`;

snapshot[`isObject 2`] = `"isObject(0): false"`;

snapshot[`isObject 3`] = `"isObject(1): false"`;

snapshot[`isObject 4`] = `"isObject(1): false"`;

snapshot[`isObject 5`] = `"isObject(Infinity): false"`;

snapshot[`isObject 6`] = `"isObject(null): false"`;

snapshot[`isObject 7`] = `"isObject(undefined): false"`;

snapshot[`isObject 8`] = `"isObject(): true"`;

snapshot[`isObject 9`] = `"isObject(1): true"`;

snapshot[`isObject 10`] = `"isObject([object Object]): true"`;

snapshot[`isObject 11`] = `"isObject([object Object]): true"`;

snapshot[`isObject 12`] = `"isObject(function Number() { [native code] }): true"`;

snapshot[`isObject 13`] = `"isObject(/fsdf/): true"`;

snapshot[`isObject 14`] = `"isObject(/fsdf/): true"`;

snapshot[`isObject 15`] = `"isObject(1.5234): false"`;

snapshot[`isObject 16`] = `"isObject(9007199254740992): false"`;

snapshot[`isObject 17`] = `"isObject(Error: hello): true"`;

snapshot[`isObject 18`] = `"isObject(NaN): false"`;

snapshot[`isObject 19`] = `"isObject(-2174774400000): false"`;

snapshot[`isObject 20`] = `"isObject([object Promise]): true"`;

snapshot[`isObject 21`] = `"isObject([object Set]): true"`;

snapshot[`isObject 22`] = `"isObject([object Map]): true"`;

snapshot[`isObject 23`] = `"isObject(/myurlorsomething): false"`;

snapshot[`isObject 24`] = `"isObject(/myregex/aa): false"`;

snapshot[`isObject 25`] = `"isObject(/myregex/): false"`;

snapshot[`isPromise 1`] = `"isPromise(s): false"`;

snapshot[`isPromise 2`] = `"isPromise(0): false"`;

snapshot[`isPromise 3`] = `"isPromise(1): false"`;

snapshot[`isPromise 4`] = `"isPromise(1): false"`;

snapshot[`isPromise 5`] = `"isPromise(Infinity): false"`;

snapshot[`isPromise 6`] = `"isPromise(null): false"`;

snapshot[`isPromise 7`] = `"isPromise(undefined): false"`;

snapshot[`isPromise 8`] = `"isPromise(): false"`;

snapshot[`isPromise 9`] = `"isPromise(1): false"`;

snapshot[`isPromise 10`] = `"isPromise([object Object]): false"`;

snapshot[`isPromise 11`] = `"isPromise([object Object]): false"`;

snapshot[`isPromise 12`] = `"isPromise(function Number() { [native code] }): false"`;

snapshot[`isPromise 13`] = `"isPromise(/fsdf/): false"`;

snapshot[`isPromise 14`] = `"isPromise(/fsdf/): false"`;

snapshot[`isPromise 15`] = `"isPromise(1.5234): false"`;

snapshot[`isPromise 16`] = `"isPromise(9007199254740992): false"`;

snapshot[`isPromise 17`] = `"isPromise(Error: hello): false"`;

snapshot[`isPromise 18`] = `"isPromise(NaN): false"`;

snapshot[`isPromise 19`] = `"isPromise(-2174774400000): false"`;

snapshot[`isPromise 20`] = `"isPromise([object Promise]): true"`;

snapshot[`isPromise 21`] = `"isPromise([object Set]): false"`;

snapshot[`isPromise 22`] = `"isPromise([object Map]): false"`;

snapshot[`isPromise 23`] = `"isPromise(/myurlorsomething): false"`;

snapshot[`isPromise 24`] = `"isPromise(/myregex/aa): false"`;

snapshot[`isPromise 25`] = `"isPromise(/myregex/): false"`;

snapshot[`isRegExp 1`] = `"isRegExp(s): false"`;

snapshot[`isRegExp 2`] = `"isRegExp(0): false"`;

snapshot[`isRegExp 3`] = `"isRegExp(1): false"`;

snapshot[`isRegExp 4`] = `"isRegExp(1): false"`;

snapshot[`isRegExp 5`] = `"isRegExp(Infinity): false"`;

snapshot[`isRegExp 6`] = `"isRegExp(null): false"`;

snapshot[`isRegExp 7`] = `"isRegExp(undefined): false"`;

snapshot[`isRegExp 8`] = `"isRegExp(): false"`;

snapshot[`isRegExp 9`] = `"isRegExp(1): false"`;

snapshot[`isRegExp 10`] = `"isRegExp([object Object]): false"`;

snapshot[`isRegExp 11`] = `"isRegExp([object Object]): false"`;

snapshot[`isRegExp 12`] = `"isRegExp(function Number() { [native code] }): false"`;

snapshot[`isRegExp 13`] = `"isRegExp(/fsdf/): true"`;

snapshot[`isRegExp 14`] = `"isRegExp(/fsdf/): true"`;

snapshot[`isRegExp 15`] = `"isRegExp(1.5234): false"`;

snapshot[`isRegExp 16`] = `"isRegExp(9007199254740992): false"`;

snapshot[`isRegExp 17`] = `"isRegExp(Error: hello): false"`;

snapshot[`isRegExp 18`] = `"isRegExp(NaN): false"`;

snapshot[`isRegExp 19`] = `"isRegExp(-2174774400000): false"`;

snapshot[`isRegExp 20`] = `"isRegExp([object Promise]): false"`;

snapshot[`isRegExp 21`] = `"isRegExp([object Set]): false"`;

snapshot[`isRegExp 22`] = `"isRegExp([object Map]): false"`;

snapshot[`isRegExp 23`] = `"isRegExp(/myurlorsomething): false"`;

snapshot[`isRegExp 24`] = `"isRegExp(/myregex/aa): false"`;

snapshot[`isRegExp 25`] = `"isRegExp(/myregex/): false"`;

snapshot[`isRegExpString 1`] = `"isRegExpString(s): false"`;

snapshot[`isRegExpString 2`] = `"isRegExpString(0): false"`;

snapshot[`isRegExpString 3`] = `"isRegExpString(1): false"`;

snapshot[`isRegExpString 4`] = `"isRegExpString(1): false"`;

snapshot[`isRegExpString 5`] = `"isRegExpString(Infinity): false"`;

snapshot[`isRegExpString 6`] = `"isRegExpString(null): false"`;

snapshot[`isRegExpString 7`] = `"isRegExpString(undefined): false"`;

snapshot[`isRegExpString 8`] = `"isRegExpString(): false"`;

snapshot[`isRegExpString 9`] = `"isRegExpString(1): false"`;

snapshot[`isRegExpString 10`] = `"isRegExpString([object Object]): false"`;

snapshot[`isRegExpString 11`] = `"isRegExpString([object Object]): false"`;

snapshot[`isRegExpString 12`] = `"isRegExpString(function Number() { [native code] }): false"`;

snapshot[`isRegExpString 13`] = `"isRegExpString(/fsdf/): false"`;

snapshot[`isRegExpString 14`] = `"isRegExpString(/fsdf/): false"`;

snapshot[`isRegExpString 15`] = `"isRegExpString(1.5234): false"`;

snapshot[`isRegExpString 16`] = `"isRegExpString(9007199254740992): false"`;

snapshot[`isRegExpString 17`] = `"isRegExpString(Error: hello): false"`;

snapshot[`isRegExpString 18`] = `"isRegExpString(NaN): false"`;

snapshot[`isRegExpString 19`] = `"isRegExpString(-2174774400000): false"`;

snapshot[`isRegExpString 20`] = `"isRegExpString([object Promise]): false"`;

snapshot[`isRegExpString 21`] = `"isRegExpString([object Set]): false"`;

snapshot[`isRegExpString 22`] = `"isRegExpString([object Map]): false"`;

snapshot[`isRegExpString 23`] = `"isRegExpString(/myurlorsomething): false"`;

snapshot[`isRegExpString 24`] = `"isRegExpString(/myregex/aa): false"`;

snapshot[`isRegExpString 25`] = `"isRegExpString(/myregex/): true"`;

snapshot[`isSet 1`] = `"isSet(s): false"`;

snapshot[`isSet 2`] = `"isSet(0): false"`;

snapshot[`isSet 3`] = `"isSet(1): false"`;

snapshot[`isSet 4`] = `"isSet(1): false"`;

snapshot[`isSet 5`] = `"isSet(Infinity): false"`;

snapshot[`isSet 6`] = `"isSet(null): false"`;

snapshot[`isSet 7`] = `"isSet(undefined): false"`;

snapshot[`isSet 8`] = `"isSet(): false"`;

snapshot[`isSet 9`] = `"isSet(1): false"`;

snapshot[`isSet 10`] = `"isSet([object Object]): false"`;

snapshot[`isSet 11`] = `"isSet([object Object]): false"`;

snapshot[`isSet 12`] = `"isSet(function Number() { [native code] }): false"`;

snapshot[`isSet 13`] = `"isSet(/fsdf/): false"`;

snapshot[`isSet 14`] = `"isSet(/fsdf/): false"`;

snapshot[`isSet 15`] = `"isSet(1.5234): false"`;

snapshot[`isSet 16`] = `"isSet(9007199254740992): false"`;

snapshot[`isSet 17`] = `"isSet(Error: hello): false"`;

snapshot[`isSet 18`] = `"isSet(NaN): false"`;

snapshot[`isSet 19`] = `"isSet(-2174774400000): false"`;

snapshot[`isSet 20`] = `"isSet([object Promise]): false"`;

snapshot[`isSet 21`] = `"isSet([object Set]): true"`;

snapshot[`isSet 22`] = `"isSet([object Map]): false"`;

snapshot[`isSet 23`] = `"isSet(/myurlorsomething): false"`;

snapshot[`isSet 24`] = `"isSet(/myregex/aa): false"`;

snapshot[`isSet 25`] = `"isSet(/myregex/): false"`;

snapshot[`isString 1`] = `"isString(s): true"`;

snapshot[`isString 2`] = `"isString(0): false"`;

snapshot[`isString 3`] = `"isString(1): false"`;

snapshot[`isString 4`] = `"isString(1): true"`;

snapshot[`isString 5`] = `"isString(Infinity): false"`;

snapshot[`isString 6`] = `"isString(null): false"`;

snapshot[`isString 7`] = `"isString(undefined): false"`;

snapshot[`isString 8`] = `"isString(): false"`;

snapshot[`isString 9`] = `"isString(1): false"`;

snapshot[`isString 10`] = `"isString([object Object]): false"`;

snapshot[`isString 11`] = `"isString([object Object]): false"`;

snapshot[`isString 12`] = `"isString(function Number() { [native code] }): false"`;

snapshot[`isString 13`] = `"isString(/fsdf/): false"`;

snapshot[`isString 14`] = `"isString(/fsdf/): false"`;

snapshot[`isString 15`] = `"isString(1.5234): false"`;

snapshot[`isString 16`] = `"isString(9007199254740992): false"`;

snapshot[`isString 17`] = `"isString(Error: hello): false"`;

snapshot[`isString 18`] = `"isString(NaN): false"`;

snapshot[`isString 19`] = `"isString(-2174774400000): false"`;

snapshot[`isString 20`] = `"isString([object Promise]): false"`;

snapshot[`isString 21`] = `"isString([object Set]): false"`;

snapshot[`isString 22`] = `"isString([object Map]): false"`;

snapshot[`isString 23`] = `"isString(/myurlorsomething): true"`;

snapshot[`isString 24`] = `"isString(/myregex/aa): true"`;

snapshot[`isString 25`] = `"isString(/myregex/): true"`;

snapshot[`isUndefined 1`] = `"isUndefined(s): false"`;

snapshot[`isUndefined 2`] = `"isUndefined(0): false"`;

snapshot[`isUndefined 3`] = `"isUndefined(1): false"`;

snapshot[`isUndefined 4`] = `"isUndefined(1): false"`;

snapshot[`isUndefined 5`] = `"isUndefined(Infinity): false"`;

snapshot[`isUndefined 6`] = `"isUndefined(null): false"`;

snapshot[`isUndefined 7`] = `"isUndefined(undefined): true"`;

snapshot[`isUndefined 8`] = `"isUndefined(): false"`;

snapshot[`isUndefined 9`] = `"isUndefined(1): false"`;

snapshot[`isUndefined 10`] = `"isUndefined([object Object]): false"`;

snapshot[`isUndefined 11`] = `"isUndefined([object Object]): false"`;

snapshot[`isUndefined 12`] = `"isUndefined(function Number() { [native code] }): false"`;

snapshot[`isUndefined 13`] = `"isUndefined(/fsdf/): false"`;

snapshot[`isUndefined 14`] = `"isUndefined(/fsdf/): false"`;

snapshot[`isUndefined 15`] = `"isUndefined(1.5234): false"`;

snapshot[`isUndefined 16`] = `"isUndefined(9007199254740992): false"`;

snapshot[`isUndefined 17`] = `"isUndefined(Error: hello): false"`;

snapshot[`isUndefined 18`] = `"isUndefined(NaN): false"`;

snapshot[`isUndefined 19`] = `"isUndefined(-2174774400000): false"`;

snapshot[`isUndefined 20`] = `"isUndefined([object Promise]): false"`;

snapshot[`isUndefined 21`] = `"isUndefined([object Set]): false"`;

snapshot[`isUndefined 22`] = `"isUndefined([object Map]): false"`;

snapshot[`isUndefined 23`] = `"isUndefined(/myurlorsomething): false"`;

snapshot[`isUndefined 24`] = `"isUndefined(/myregex/aa): false"`;

snapshot[`isUndefined 25`] = `"isUndefined(/myregex/): false"`;
